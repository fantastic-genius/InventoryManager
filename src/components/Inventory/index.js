import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

//contants
import colors from '../../constants/colors';

const Inventory = ({title, description, price, stock, onPress, testId}) => {
  return (
    <TouchableOpacity style={styles.item} onPress={onPress} testID={testId}>
      <View style={styles.content}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.description}>{description}</Text>
      </View>
      <View style={styles.rightContent}>
        <Text style={styles.price}>{`NGN${price}`}</Text>
        <Text style={styles.stock}>
          {`${stock} ${stock > 1 ? 'pieces' : 'peice'}`}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  item: {
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: colors.WHITE,
    borderRadius: 10,
    marginBottom: 16,
  },
  content: {
    paddingRight: 16,
    flex: 1,
  },
  rightContent: {
    alignItems: 'flex-end',
  },
  title: {
    fontSize: 16,
    fontWeight: '700',
    color: colors.SEMIDARKGRAY,
    lineHeight: 22,
  },
  description: {
    fontSize: 12,
    fontWeight: '400',
    color: colors.MEDIUMGRAY,
    lineHeight: 20,
  },
  price: {
    fontSize: 14,
    fontWeight: '600',
    color: colors.SEMIDARKGRAY,
    lineHeight: 20,
  },
  stock: {
    fontSize: 12,
    fontWeight: '500',
    color: colors.MEDIUMGRAY,
    lineHeight: 20,
  },
});

export default Inventory;
