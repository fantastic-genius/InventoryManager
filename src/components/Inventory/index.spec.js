import React from 'react';
import renderer from 'react-test-renderer';
import Inventory from './index';

describe('Inventory', () => {
  it('renders component as expected', async () => {
    const component = renderer.create(<Inventory />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
