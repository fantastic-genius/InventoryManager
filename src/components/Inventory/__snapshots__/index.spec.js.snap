// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory renders component as expected 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingRight": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212C3D",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 22,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#627483",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212C3D",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
        }
      }
    >
      NGNundefined
    </Text>
    <Text
      style={
        Object {
          "color": "#627483",
          "fontSize": 12,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      undefined peice
    </Text>
  </View>
</View>
`;
